### DISCUSSION TITLE
Question: how to create cross-node-type extension?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [remirror/remirror#277] [Q&A]
I want to create an alignment extension for all block nodes. A key point is not to modify the pre-defined built-in extensions.

Looking through the codebase, extraAttrs seems a way to go, but I found extensions such as heading extension didn't parse the extraAttrs. And I didn't make it to create a workable demo for this purpose.

I noticed the paragraph extension already has an attr `align`, which implements this functionality, but how to extend it to other types? Do I have to extends all types with such code from paragraph to achieve this goal?

## Checklist

- [ ] I have read the [**contributing**](https://github.com/remirror/remirror/blob/master/docs/contributing.md) document.

**Comment by ifiokjr** [comment]
Thanks for taking a look at alignment. It'd be great to have that integration. 

Based on what you've just mentioned, I'm going to look at improving the extraAttrs to automatically parse the added attributes for all nodes and marks in the `next` branch. 

Thanks for the suggestion. 

> I noticed the paragraph extension already has an attr align, which implements this functionality, but how to extend it to other types? Do I have to extends all types with such code from paragraph to achieve this goal?

I'm actually not sure. I'll look at it. If it can't be done I'll think through how it can be improved in the `next` release.

**Comment by ifiokjr** [comment]
@zhaoyao91 I've moved this over to a discussion since it seems like a better fit. 

It's possible to do this now in the `next` branch by using the `createSchemaAttributes` method and updating the attributes for multiple extensions.

There's no documentation yet for it but the `@remirror/extension-bidi` is making use of the functionality.

https://github.com/remirror/remirror/blob/d4411e5070aef9f28aef841459023c44d01ddd40/packages/%40remirror/extension-bidi/src/bidi-extension.ts#L68-L77

**Reply by anotheri** [reply]
any chance we'll see alignment extension soon?

**Reply by ifiokjr** [reply]
It's actually possible in the beta branch. 
I'll create some docs for this soon.
